Mathlib/Data/Finset/Card.lean

```diff
@@ -539,6 +539,10 @@ theorem card_le_one_iff : s.card ≤ 1 ↔ ∀ {a b}, a ∈ s → b ∈ s → a
   tauto
 #align finset.card_le_one_iff Finset.card_le_one_iff
 
+theorem card_le_one_iff_subsingleton_coe : s.card ≤ 1 ↔ Subsingleton (s : Type _) := by
+  rw [card_le_one_iff, subsingleton_iff]
+  aesop
```

Comment:
a small golf:
```suggestion
theorem card_le_one_iff_subsingleton_coe : s.card ≤ 1 ↔ Subsingleton (s : Type _) :=
  card_le_one.trans (s : Set α).subsingleton_coe.symm
```
which also makes it easier to find the corresponding `Set` lemma.